src/wfs/src/main/java/org/geoserver/wfs/xml/FeatureTypeSchemaBuilder.java

```diff
@@ -780,8 +780,14 @@ private XSDComplexTypeDefinition buildComplexSchemaContent(ComplexType complexTy
                         Class binding = attribute.getType().getBinding();
                         typeName = findTypeName(binding);
                         if (typeName == null) {
-                            throw new NullPointerException("Could not find a type for property: "
-                                    + attribute.getName() + " of type: " + binding.getName());
+                            // Fallback on String
+                            logger.finer("Fallback mapping: attribute " + attribute.getName() + " to String" );
+                            typeName = findTypeName(String.class);
+                            if (typeName == null) {
+                                throw new NullPointerException(
```

Comment:
Maybe a WFSException or a ServiceException instead?

Reply:
I have preferred to preserve the original exception (see old line 783) in case any caller is catching that specific exception.